### PR TITLE
Update Github Actions

### PR DESCRIPTION
**Pull request by PhMueller**
Update the github actions. 

- [x] Remove support for older singularity versions 
- [x] Add support forsingularity 3.9 and 3.10

### CODE REVIEW

**Review by PhMueller — COMMENTED** [review]
Submitting a feedback now triggers a github action workflow.

**Review by PhMueller — COMMENTED** [review]
Trigger Event 2nd Time

**Review by PhMueller — COMMENTED** [review]
Trigger a 3rd time

File: README.md [hunk 1]
```diff
@@ -54,14 +54,14 @@ cd HPOBench
 pip install .
 ```
 
-**Note:** This does not install *singularity (version 3.6)*. Please follow the steps described here: [user-guide](https://sylabs.io/guides/3.6/user-guide/quick_start.html#quick-installation-steps).   
+**Note:** This does not install *singularity (version 3.8)*. Please follow the steps described here: [user-guide](https://sylabs.io/guides/3.8/user-guide/quick_start.html#quick-installation-steps).
```
Comment (KEggensperger) [inline README.md:57]: Also update text for troubleshooting at the end of the readme

File: ci_scripts/install_singularity.sh [hunk 2]
```diff
@@ -14,21 +14,33 @@ sudo apt-get update && sudo apt-get install -y \
...
-    export VERSION=3.6.4
-elif [[ "$SINGULARITY_VERSION" == "3.7" ]]; then
+if [[ "$SINGULARITY_VERSION" == "3.7" ]]; then
     export VERSION=3.7.3
+    export FILENAME=singularity-"${VERSION}"
+    export EXTRACTED_FILENAME=singularity
+
 elif [[ "$SINGULARITY_VERSION" == "3.8" ]]; then
-    export VERSION=3.8.0
+    export VERSION=3.8.4
```
Comment (KEggensperger) [inline ci_scripts/install_singularity.sh:23]: Is there a specific reason for exactly this version?

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,3 @@
-scipy>=1.4.1
```
Comment (KEggensperger) [inline requirements.txt:1]: What's the reason for removing this dependency?

**Review by KEggensperger — APPROVED** [review]

File: ci_scripts/install_singularity.sh (hunk 2 above)
Comment (PhMueller) [reply ci_scripts/install_singularity.sh:23]: Yes, I always use the "newest" version per release. 
For the 3.7 -> 3.7.3, for 3.8 -> 3.8.4 and so on.

File: README.md (hunk 1 above)
Comment (PhMueller) [reply README.md:57]: Actually, this issue (**Singularity throws an 'Invalid Image format' exception**) is outdated. On the slurm cluster, we have only 3.9 now

File: requirements.txt (hunk 3 above)
Comment (PhMueller) [reply requirements.txt:1]: It is only a dependency for the old XGBoost Benchmark and we don't need it as "general" dependency in the HPOBench. Furthermore, it is a direct dependency of the ConfigSpace. So, I think it is okay to remove here.

**Pull request merged**